product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.10-Setting-a-request-payload-with-a-script/src/test/java/org.wso2.carbon.ei.scenario.test/AlterRequestWithScriptTest.java

```diff
@@ -0,0 +1,132 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.ei.scenario.test;
+
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.esb.scenario.test.common.http.HTTPUtils;
+import org.wso2.carbon.esb.scenario.test.common.ScenarioTestBase;
+
+import java.io.IOException;
+import javax.xml.stream.XMLStreamException;
+
+/**
+ * This class is to test if xml payload can be enriched before it goes to the backend server by using script mediator.
+ **/
+public class AlterRequestWithScriptTest extends ScenarioTestBase {
+
+     private  static final String REQUEST_1_6_10 =
+             "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+                     + "<soap:Envelope xmlns:soap=\"http://schemas.xmlsoap.org/soap/envelope/\">\n"
+                     + "   <soap:Header />\n"
+                     + "   <soap:Body>\n"
+                     + "      <employees>\n"
+                     + "         <employee>\n"
+                     + "            <age>25</age>\n"
+                     + "            <firstName>John</firstName>\n"
+                     + "            <lastName>Doe</lastName>\n"
+                     + "         </employee>\n"
+                     + "         <employee>\n"
+                     + "            <age>45</age>\n"
+                     + "            <firstName>Anna</firstName>\n"
+                     + "            <lastName>Smith</lastName>\n"
+                     + "         </employee>\n"
+                     + "         <employee>\n"
+                     + "            <age>35</age>\n"
+                     + "            <firstName>Peter</firstName>\n"
+                     + "            <lastName>Jones</lastName>\n"
+                     + "         </employee>\n"
+                     + "      </employees>\n"
+                     + "   </soap:Body>\n"
+                     + "</soap:Envelope>";
+
+     @BeforeClass
+     public void init() throws Exception {
+        super.init();
+     }
+
+     /**
+      * This test is to verify if payload can be modified by removing first element using inline groovy script.
+      * This testcase has been disabled since we need to manually paste the groovy-all-dependency jar into $EI_HOME/lib.
+      */
+    @Test(description = "1.6.10.1", enabled=false)
+    public void alterPayloadByInlineGroovyScript() throws IOException, XMLStreamException {
```

Comment:
Why are we adding a test case that is not enabled?

Reply:
Because the groovy all dependency jar needs to be added manually.